src/cmap/connection_pool.ts

```diff
@@ -219,6 +235,31 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
     return this[kWaitQueue].length;
   }
 
+  get loadBalanced(): boolean {
+    return this.options.loadBalanced;
+  }
+
+  get serviceGenerations(): Map<ObjectId, number> {
+    return this[kServiceGenerations];
+  }
+
+  markPinned(connection: Connection, pinType: string): void {
+    connection.markPinned(pinType);
+    this[kMetrics].markPinned(pinType);
+  }
+
+  markUnpinned(connection: Connection, pinType: string): void {
+    connection.markUnpinned(pinType);
+    this[kMetrics].markUnpinned(pinType);
+  }
+
+  /**
+   * Get the metrics information for the pool when a wait queue timeout occurs.
+   */
+  waitQueueErrorMetrics(): string {
```

Comment:
Wait queue timeouts in LB mode have a more detailed error message on pinned connections.

Reply:
Sorry to revisit, can we use `private` instead of underscore here? We can make use of typescript to prevent usage outside this class